service/common/src/main/java/org/apache/polaris/service/events/BeforeTableRefreshEvent.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.polaris.service.events;
+
+import org.apache.iceberg.catalog.TableIdentifier;
+
+/**
+ * Emitted when Polaris intends to refresh its known version of a table's metadata by fetching the
+ * latest.
+ *
+ * @param tableIdentifier The identifier of the table being refreshed.
+ */
+public record BeforeTableRefreshEvent(TableIdentifier tableIdentifier) {}
```

Comment:
The events don't have a shared parent type?

Reply:
Curious what your reasons are for wanting this.

I could see at least 2 reasons for having a shared type:
1. Having a more specific parent class than Object to know that you're dealing with a set of events
2. Leaving room for fields in the future that could be common to all events

I'm not sure when those would be useful as each event is different. In the future we could always make all the records implement an interface.

Reply:
Yea I just mean currently the data in them is totally different. Even in Spark, the only common method is `logEvent()` which arguably isn't worth an entire interface. I added a marker one for now.